crates/miden-objects/src/account/component/storage/init_storage_data.rs

```diff
@@ -0,0 +1,79 @@
+use alloc::collections::BTreeMap;
+use alloc::string::String;
+use alloc::vec::Vec;
+
+use super::StorageValueName;
+
+/// A raw word value provided via [`InitStorageData`].
+///
+/// This is used for map entries, where keys and values are supplied as either a scalar string
+/// (e.g. `"0x1234"`, `"16"`, `"BTC"`) or an array of 4 scalar elements.
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "std", serde(untagged))]
+pub enum WordValue {
+    Scalar(String),
+    Elements([String; 4]),
+}
+
+impl From<String> for WordValue {
+    fn from(value: String) -> Self {
+        WordValue::Scalar(value)
+    }
+}
+
+impl From<&str> for WordValue {
+    fn from(value: &str) -> Self {
+        WordValue::Scalar(String::from(value))
+    }
+}
+
+/// Represents the data required to initialize storage entries when instantiating an
+/// [AccountComponent](crate::account::AccountComponent) from component metadata (either provided
+/// directly or extracted from a package).
+///
+/// An [`InitStorageData`] can be created from a TOML string when the `std` feature flag is set.
+#[derive(Clone, Debug, Default)]
+pub struct InitStorageData {
+    /// A mapping of init value names to their raw values.
+    value_entries: BTreeMap<StorageValueName, WordValue>,
+    /// A mapping of storage map slot names to their raw key/value entries.
+    map_entries: BTreeMap<StorageValueName, Vec<(WordValue, WordValue)>>,
+}
```

Comment:
Should the keys in these maps be `StorageSlotName`? I'm assuming that each `WordValue` represents a single word - and so, we won't have multiple entries for the same storage slot in these maps.

Maybe for a different PR, but should also consider combining these into a single map - e.g.:

```rust
pub struct InitStorageData {
    entries: BTreeMap<StorageSlotName, StorageSlotEntry>,
}

pub enum StorageSlotEntry {
    Value(WordValue),
    Map(Vec<(WordValue, WordValue)>),
}
```

Reply:
Added this to #1860 as something to keep in mind